ConsumerAffairs/django-experiments
# RED-8 experiments admin panel

## Overview

This PR implements read-only part of the experiments dashboard feature. The feature to make changes will be part of another PR.

Dashboard is implemented in the admin, on a regular changelist admin view. 

## Details

## Testing instructions

Freely ping `fhrzenjak` on Slack to assist with practicals.



## Tests

- [x] Test 1. - check that the API is up
1. Log in as an admin user
2. Open `/experiments/api/`, and verify that:
    * "browsable API" interface is rendered
    * the representation contains a link to `v1` of the API
3. Click on the link to `v1`, and verify that:
    * this representation contains a link to list of all experiments on this site
4. Click on the link, and verify that:
    * the results really contain all experiments on the site
5. Click on link under a`url` field, and verify that:
    * the representation shows only the selected experiment
    * `state` of the experiment can be changed using the form for `PUT` requests at the bottom of the page

- [x] Test 2. - check that admin dashboard is up
1. Open `/admin/experiments/remoteexperiment/` and verify that:
    * all the experiments on the site are listed here
    * if you are testing with multiple sites configured, the list should contain experiments from all sites:
        
    * at this point, all `confidence intervals` should be either `no alternatives` or `no primary goals`
2. Open an experiment with says `no primary goals`
3. Scroll down to `Primary Goals` section (on the main additionally click `expand` link to see the table)
4. Mark `Pri` checkbox (stands for "primary") next to one of already marked goals.
   
5. Click `Save and continue editing` and verify that:
    * in the "Results" table (at the top of the page) the selected goal in now marked with bold fontface
    * take note of the confidence interval values for that goal

6. Go back to `/admin/experiments/remoteexperiment/`, reload, and verify that:
    * the edited experiment now shows a mini table under `confidence intervals` with values from previous step
